src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -246,13 +246,18 @@ def update_resolvable_node(
         # update the pre_actions and post_actions for the select statement
         node.pre_actions = node.from_.pre_actions
         node.post_actions = node.from_.post_actions
-        node.expr_to_alias = node.from_.expr_to_alias
+
```

Comment:
for a `SelectStatement`, node.expr_to_alias is not necessarily derived from child selectable. The happy path to update `expr_to_alias` for `SnowflakePlan` is during resolve step at https://github.com/snowflakedb/snowpark-python/blob/47ee32c51b6ea1914ffcd6a8a541949918a18cd0/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L863 

This value is propagates to `SelectStatement` at https://github.com/snowflakedb/snowpark-python/blob/47ee32c51b6ea1914ffcd6a8a541949918a18cd0/src/snowflake/snowpark/dataframe.py#L605